sympy/matrices/matrices.py

```diff
@@ -1153,7 +1153,10 @@ def eigenvals(self, error_when_incomplete=True, **flags):
                     eigs[diagonal_entry] += 1
         else:
             flags.pop('simplify', None)  # pop unsupported flag
-            eigs = roots(mat.charpoly(x=Dummy('x'), simplify=simplify if isinstance(simplify, FunctionType) else _simplify), **flags)
+            if isinstance(simplify, FunctionType):
+                eigs = roots(mat.charpoly(x=Dummy('x'), simplify=simplify), **flags)
+            else:
+                eigs = roots(mat.charpoly(x=Dummy('x'), simplify=_simplify), **flags)
```

Comment:
We could leave `simplify` undefined. That would make it possible to change the default of `charpoly` with no modifications needed here.